docs/testing/02-cli-tests.mdx

```diff
@@ -0,0 +1,123 @@
+---
+sidebar_position: 2
+sidebar_label: "CLI Tests"
+id: kuneiform-cli-tests
+title: CLI tests
+description: Testing Kuneiform with `kwil-cli`
+slug: /testing/kwil-cli
+---
+
+To run CLI tests, you will need the `kwil-cli` binary installed. To download the binary,
+see the [downloads page of Github](<https://github.com/kwilteam/kwil-db/releases>).
+
+This documentation explains how to define JSON tests using an example Kuneiform
+schema. The schema is a basic application that stores and retrieves users:
+
+<details>
+  <summary> users.kf </summary>
+
+```sql showLineNumbers
+database users_db;
+
+table users {
+  id uuid primary key,
+  name text not null unique,
+  address text not null unique
+}
+
+procedure create_user ($name text) public  {
+  // derive a deterministic uuid from the blockchain transaction ID
+  // https://www.postgresql.org/docs/16.1/uuid-ossp.html#UUID-OSSP-FUNCTIONS-SECT
+  $uuid := uuid_generate_v5('f541de32-5ede-4083-bdbc-b29c3f02be9e'::uuid, @txid);
+
+  insert into users (id, name, address)
+  values ($uuid, $name, @caller);
+}
+
+procedure get_users() public view returns table (name text, address text) {
+  return select name, address from users;
+}
+```
+</details>
+
+## Defining A Test
+
+To define a test, create a new JSON file called `tests.json`. This test assumes we have the current directory
+setup:
+
+```shell
+working-dir/
+    |- tests.json
+    |- users.kf
+```
+
+The below file gives our tests a top-level name of "my users.kf tests". It will read in
+the schema in the file `users.kf`. It defines an insert statement to seed data into the
```

Comment:
Do `seed_statements` run before all tests? If so, may be appropriate to call out.

Reply:
Yes they do. Will call that out.